src/java/org/apache/cassandra/db/rows/UnfilteredRowIteratorWithLowerBound.java

```diff
@@ -87,10 +86,27 @@ public Unfiltered lowerBound()
         if (lowerBoundMarker != null)
             return lowerBoundMarker;
 
-        // The partition index lower bound is more accurate than the sstable metadata lower bound but it is only
-        // present if the iterator has already been initialized, which we only do when there are tombstones since in
-        // this case we cannot use the sstable metadata clustering values
-        lowerBound = getMetadataLowerBound();
+        if (initialized())
+        {
+            // The partition index lower bound is more accurate than the sstable metadata lower bound but it is only
+            // present if the iterator has already been initialized
+            lowerBound = getKeyCacheLowerBound();
```

Comment:
This path should be used for both cases after making sure `getKeyCacheLowerBound` never initializes the iterator.

Also, do we need to store both the bound and the marker? If we need it, the bound can be retrieved from the marker.

Reply:
ok